api/src/main/java/org/openmrs/module/bedmanagement/aop/EncounterWithBedPatientAssignmentSaveHandler.java

```diff
@@ -0,0 +1,68 @@
+/**
+ * The contents of this file are subject to the OpenMRS Public License
+ * Version 1.0 (the "License"); you may not use this file except in
+ * compliance with the License. You may obtain a copy of the License at
+ * http://license.openmrs.org
+ *
+ * Software distributed under the License is distributed on an "AS IS"
+ * basis, WITHOUT WARRANTY OF ANY KIND, either express or implied. See the
+ * License for the specific language governing rights and limitations
+ * under the License.
+ *
+ * Copyright (C) OpenMRS, LLC.  All Rights Reserved.
+ */
+package org.openmrs.module.bedmanagement.aop;
+
+import java.util.Date;
+import java.util.List;
+
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.openmrs.Encounter;
+import org.openmrs.User;
+import org.openmrs.annotation.Handler;
+import org.openmrs.api.handler.SaveHandler;
+import org.openmrs.module.bedmanagement.entity.BedPatientAssignment;
+import org.openmrs.module.bedmanagement.service.BedManagementService;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Qualifier;
+
+@Handler(supports = Encounter.class)
+public class EncounterWithBedPatientAssignmentSaveHandler implements SaveHandler<Encounter> {
+	
+	private final Log log = LogFactory.getLog(getClass());
+	
+	private final BedManagementService bedManagementService;
+	
+	@Autowired
+	public EncounterWithBedPatientAssignmentSaveHandler(
+	    @Qualifier("bedManagementService") BedManagementService bedManagementService) {
+		this.bedManagementService = bedManagementService;
+	}
+	
+	@Override
+	public void handle(Encounter encounter, User user, Date date, String s) {
+		
+		if (encounter.getEncounterId() != null) {
+			
+			List<BedPatientAssignment> bpaList = bedManagementService.getBedPatientAssignmentByEncounter(encounter.getUuid(),
```

Comment:
Is there a use case for more than one encounter to be linked to the same bed assignment?  (I'm not suggesting this code needs to be changed, just wanted to make sure I understand the use cases properly).

Reply:
AFAIK, nothing prevent an encounter from being reused for multiple bed assignments. That said, the O3 ward app always creating a bed assigning encounter for each new bed assignment, and I think that's the case as well for the O2 Bahmni UI.